hawkular-agent-installer/src/main/java/org/hawkular/agentinstaller/AgentInstaller.java

```diff
@@ -0,0 +1,169 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.io.IOUtils;
+import org.hawkular.extensioninstaller.DeploymentConfiguration;
+import org.hawkular.extensioninstaller.ExtensionDeployer;
+import org.hawkular.extensioninstaller.XmlEdit;
+import org.jboss.logging.Logger;
+
+public class AgentInstaller {
+
+    private static final Logger log = Logger.getLogger(AgentInstaller.class);
+    private static final String OPTION_WILDFLY_HOME="wildfly-home";
+    private static final String OPTION_MODULE = "module";
+    private static final String OPTION_HAWKULAR_SERVER_URL = "hawkular-server-url";
+
+    private static final Options OPTIONS = commandLineOptions();
+
+    public static void main(String[] args) throws Exception {
+        try {
+            CommandLine commandLine = new DefaultParser().parse(OPTIONS, args);
+
+            String jbossHome = commandLine.getOptionValue(OPTION_WILDFLY_HOME);
+            String moduleZip = commandLine.getOptionValue(OPTION_MODULE);
+            String hawkularServerUrl = commandLine.getOptionValue(OPTION_HAWKULAR_SERVER_URL);
+
+            File moduleFile = null;
+            // if --module is not supplied try to download agent module from server
+            if (moduleZip == null) {
+                moduleFile = downloadModule(hawkularServerUrl);
+                if (moduleFile == null) {
+                    throw new ParseException("Failed to retrieve agent module from server, option [module]"
+                            + " is now required but it was not supplied");
+                }
+            } else {
+                moduleFile = new File(moduleZip);
+            }
+
+            // deploy given module into given wfHome and
+            // set it up the way it talks to hawkular server on hawkularServerUrl
+            // TODO support domain mode
+            // TODO support non default server config (ie standalone-full.xml)
+            DeploymentConfiguration configuration = DeploymentConfiguration.builder()
+                    .jbossHome(new File(jbossHome))
+                    .module(moduleFile.toURI().toURL())
+                    .addXmlEdit(replaceServerUrl(hawkularServerUrl))
+                    .build();
+            new ExtensionDeployer().install(configuration);
+        } catch (ParseException pe) {
+            log.warn(pe);
+            help();
+        }
+    }
+
+    /**
+     * creates XmlEdit which changes "<storage-adapter" url attribute to "hawkluarServerUrl"
+     * @param hawkularServerUrl
+     * @return
+     */
+    private static XmlEdit replaceServerUrl(String hawkularServerUrl) {
+        String select = "/server/profile/"
+                + "*[namespace-uri()='urn:org.hawkular.agent.monitor:monitor:1.0']/";
+
+        StringBuilder xml = new StringBuilder("<storage-adapter")
+            .append(" type=\"HAWKULAR\"")
+            .append(" url=\""+hawkularServerUrl+"\"")
```

Comment:
we do NOT want to use "url" attribute (I'm gonna get rid of this soon since everyone keep using it :)) - we want to use the standard binding mechanism that WF provides; putting in a url with custom ports violates standard practice in WF subsystems. Instead, we want to add a &lt;outbound-socket-binding that defines the host/port to the hawkular server:

&lt;outbound-socket-binding name="hawkular">
  <&lt;remote-destination host="your-hawkular-server-hostname"  port="the-port" />
&lt;/outbound-socket-binding>

This is why i ask if we can use this tool to add extra stuff in standalone.xml. We need to add this outbound socket binding, and optionally the security-realm and https-listener.